translator/translate/logs/logs_collected/files/collect_list/collect_list_test.go

```diff
@@ -611,3 +611,115 @@ func TestLogFilters(t *testing.T) {
 	}}
 	assert.Equal(t, expectVal, val)
 }
+
+func TestRetention(t *testing.T) {
+	f := new(FileConfig)
+	var input interface{}
+	e := json.Unmarshal([]byte(`{
+		"collect_list":[
+			{
+				"file_path":"path1",
+       			"log_group_name":"test2",
+				"retention_in_days":3
+			},
+			{
+				"file_path":"path1",
+       			"log_group_name":"test1",
+				"retention_in_days":3
+			}
+		]
+	}`), &input)
+	if e != nil {
+		assert.Fail(t, e.Error())
+	}
+	_, val := f.ApplyRule(input)
+	expectVal := []interface{}{map[string]interface{}{
+		"file_path":         "path1",
+		"log_group_name":    "test2",
+		"pipe":              false,
+		"retention_in_days": 3,
+		"from_beginning":    true,
+	}, map[string]interface{}{
+		"file_path":         "path1",
+		"log_group_name":    "test1",
+		"pipe":              false,
+		"retention_in_days": 3,
+		"from_beginning":    true,
+	}}
+	assert.Equal(t, expectVal, val)
+}
+
+func TestDuplicateRetention(t *testing.T) {
+	f := new(FileConfig)
+	var input interface{}
+	e := json.Unmarshal([]byte(`{
+		"collect_list":[
+			{
+				"file_path":"path1",
+       			"log_group_name":"test1",
+				"retention_in_days":3
+			},
+			{
+				"file_path":"path1",
+       			"log_group_name":"test1",
+				"retention_in_days":3
+			}
+		]
+	}`), &input)
+	if e != nil {
+		assert.Fail(t, e.Error())
+	}
+	_, val := f.ApplyRule(input)
+	expectVal := []interface{}{map[string]interface{}{
+		"file_path":         "path1",
+		"log_group_name":    "test1",
+		"pipe":              false,
+		"retention_in_days": 3,
+		"from_beginning":    true,
+	}, map[string]interface{}{
+		"file_path":         "path1",
+		"log_group_name":    "test1",
+		"pipe":              false,
+		"retention_in_days": -1,
+		"from_beginning":    true,
+	}}
+	assert.Equal(t, expectVal, val)
+}
+
+func TestConflictingRetention(t *testing.T) {
+	f := new(FileConfig)
+	var input interface{}
+	e := json.Unmarshal([]byte(`{
+		"collect_list":[
+			{
+				"file_path":"path1",
+       			"log_group_name":"test1",
+				"retention_in_days":3
+			},
+			{
+				"file_path":"path1",
+       			"log_group_name":"test1",
+				"retention_in_days":5
+			}
+		]
+	}`), &input)
+	if e != nil {
+		assert.Fail(t, e.Error())
+	}
```

Comment:
Could change:
```
	if e != nil {
		assert.Fail(t, e.Error())
	}
```
to:
```
assert.Equal(e, WhateverTheExpectedErrorIs)
```
It would be more specific.